client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -93,24 +93,32 @@
         private long eventNumber = 0;
         private final ReusableLatch connectionSetup = new ReusableLatch();
         @GuardedBy("lock")
-        private CompletableFuture<Void> emptyInflightFuture = null;
+        private CompletableFuture<Exception> emptyInflightFuture = null;
 
         /**
          * Returns a future that will complete successfully once all the inflight events are acked
          * and will fail if that is not possible for some reason. IE: because the connection
          * dropped, or the segment was sealed.
          */
-        private CompletableFuture<Void> getEmptyInflightFuture() {
+        private CompletableFuture<Exception> getEmptyInflightFuture() {
```

Comment:
This code is all private and it doesn't matter that much if we abuse the language features a bit, but semantically it does look somewhat odd. The future type should be an expected value as opposed to an exception, which is supposed to be that, an exceptional condition.

Again, I don't find it a super big deal, but I'd like to understand what is annoying about the unwrapping, even if it is just for my own education.